internal/bminventory/inventory.go

```diff
@@ -2062,6 +2066,15 @@ func (b *bareMetalInventory) updateNetworkParams(params installer.UpdateClusterP
 		return common.NewApiError(http.StatusBadRequest, err)
 	}
 
+	if err = validations.ValidateNetworkTypeWithIPv6(params.ClusterUpdateParams.NetworkType, machineCidr); err != nil {
+		validNetworkTypes := []string{models.ClusterNetworkTypeOVNKubernetes, models.ClusterNetworkTypeOpenShiftSDN}
+		if !funk.ContainsString(validNetworkTypes, swag.StringValue(params.ClusterUpdateParams.NetworkType)) {
```

Comment:
I would expect to first check that the type itself is valid, then check if it's compatible with the machineCIDR

Reply:
Not just machineCIDR,but with serviceCIDR and clusterCIDR